sys/include/net/ieee802154.h

```diff
@@ -46,6 +46,8 @@ extern "C" {
  * @{
  */
 #define IEEE802154_MAX_HDR_LEN              (23U)
+#define IEEE802154_MIN_FRAME_LEN            (3)
```

Comment:
Mh, to make clearer where this number is coming from, maybe make the value `IEEE802154_FCF_LEN + sizeof(uint8_t)`? If not, it should at least be `3U`.

Reply:
> IEEE802154_FCF_LEN + sizeof(uint8_t)

Sounds good.

> If not, it should at least be 3U

Make sense but since `bytes_expected` is a signed value we need to cast `IEEE802154_MIN_FRAME_LEN` to a signed value as well or cast `bytes_expected` to an unsigned value in gnrc_netif_ieee802154.c` `due to the fact that RIOT uses `-Wsign-compare`.

Reply:
> Make sense but since bytes_expected is a signed value we need to cast IEEE802154_MIN_FRAME_LEN to a signed value as well or cast bytes_expected to an unsigned value in gnrc_netif_ieee802154.c due to the fact that RIOT uses -Wsign-compare. How should I handle this?

I'm not sure, but I seem to remember that `signed < unsigned` (which is what we are doing) should be fine.

Reply:
(it still makes logical sense at least).

Reply:
Well I am currently comparing signed >= unsigned which fails with:

```
/root/RIOT/sys/net/gnrc/netif/gnrc_netif_ieee802154.c: In function '_recv':
/root/RIOT/sys/net/gnrc/netif/gnrc_netif_ieee802154.c:85:24: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
     if (bytes_expected >= IEEE802154_MIN_FRAME_LEN) {
                        ^~
```

I just pushed this and would simply change it to `

```C
/* ... */
if (bytes_expected >= (int)IEEE802154_MIN_FRAME_LEN) {
/* ... */
```
 
If you are ok with that…